lib/tasks/demo_data_for_development.rake

```diff
@@ -0,0 +1,70 @@
+namespace :data do
+  desc 'Create demo data for our local development'
+  task test: :environment do
+    include FactoryGirl::Syntax::Methods
+
+    def generate_events program, conference
+        user1 = create(:user)
+        user2 = create(:user)
+
+        selected_schedule = create(:schedule, program: program)
+        demo_schedule = create(:schedule, program: program)
+        program.update_attributes(selected_schedule: selected_schedule)
+
+        build(:event, program: program, title: 'Demo Event', abstract: 'This is a demo event instance whose state not defined.').save(validate: false)
```

Comment:
@siddhantbajaj
 > One possible solution is we change the validation and compare end_date with created_at attribute of the event instead of Date.today.

Do not change validation but the way you create the events to  make then valid. You could change the conference start time to today, create the events and then update the conference dates, right?